pkg/kubelet/im/image_gc.go

```diff
@@ -27,23 +27,12 @@ import (
 	"k8s.io/kubernetes/pkg/client/record"
 	"k8s.io/kubernetes/pkg/kubelet/cadvisor"
 	"k8s.io/kubernetes/pkg/kubelet/container"
-	"k8s.io/kubernetes/pkg/util"
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-// Manages lifecycle of all images.
-//
-// Implementation is thread-safe.
-type imageManager interface {
-	// Applies the garbage collection policy. Errors include being unable to free
-	// enough space as per the garbage collection policy.
-	GarbageCollect() error
-
-	// Start async garbage collection of images.
-	Start() error
-
-	// TODO(vmarmol): Have this subsume pulls as well.
-}
+const (
+	defaultGCAge = time.Minute * 5
```

Comment:
One more option to consider:
1. Sort all the images based on the creation time and size (Last access time might be more useful in reality).
2. Remove as many images as necessary to bring down disk utilization.

Even with this mode there is a possibility for removing a large image that just got pulled. By giving more weight for age instead of size, we can try to mitigate that issue.

We are also working on improving disk accounting and eventually rotate logs aggressively, or evict containers if necessary. Once those features land, image GC should be less of a concern.